File: P415 Add Strings/Solution.cpp
class Solution {
public:
    string addStrings(string num1, string num2) {
        int len1 = num1.length() - 1;
        int len2 = num2.length() - 1;
        int carry = 0;
        string res = "";
        while(len1>=0 || len2>=0 || carry == 1){
            long sum = 0;
            if(len1 >= 0){
                sum += (num1[len1] - '0');
                len1--;
            }
            if(len2 >= 0){
                sum += (num2[len2] - '0');
                len2--;
            }
            sum += carry; 
            carry = (int) sum / 10;
            sum = sum % 10;
            res = to_string(sum) + res;
        }
        return res;
    }
};

File: P415 Add Strings/Solution.py
class Solution(object):
    def addStrings(self, num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
        len1 = len(num1) - 1
        len2 = len(num2) - 1
        carry = 0
        res = ""
        while(len1>=0 or len2>=0 or carry):
            sum = 0
            if(len1 >= 0):
                sum += (ord(num1[len1]) - 48)
                len1 -= 1
            if(len2 >= 0):
                sum += (ord(num2[len2]) - 48)
                len2 -= 1
            sum += carry
            carry = sum / 10
            sum = sum % 10
            res = str(sum) + res
        return res